src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml

```diff
@@ -0,0 +1,82 @@
+            <Setter Property="AutomationProperties.Name" Value="{Binding AutomationName}"/>
+            <Setter Property="IsCheckable" Value="False"/>
+            <Setter Property="Foreground" Value="{DynamicResource {x:Static vsui:EnvironmentColors.CommandBarTextActiveBrushKey}}"/>
+            <Setter Property="Background" Value="Transparent"/>
+            <EventSetter Event="Click" Handler="TargetMenuItem_OnClick"/>
+        </Style>
+
+        <!-- The key is referenced from the code behind -->
+        <!-- Style used to display a single member's information, if click the button, it pops up a menu contains targets' information (one level menu)-->
+        <Style x:Key="SingleMemberContextMenuStyle" TargetType="{x:Type ContextMenu}">
+            <Setter Property="ItemContainerStyle" Value="{StaticResource TargetMenuItemStyle}"/>
+            <Setter Property="ItemsSource" Value="{Binding Targets}"/>
+            <Setter Property="StaysOpen" Value="True"/>
+        </Style>
+
+        <!-- Style used to display one member's information and pops up a submenu show all the targets -->
+        <Style x:Key="MemberMenuItemStyle" TargetType="{x:Type MenuItem}">
+            <Setter Property="Icon" Value="{StaticResource NonSharedIcon}"/>
+            <Setter Property="Header" Value="{Binding DisplayName}"/>
+            <Setter Property="ToolTip" Value="{Binding ToolTip}"/>
+            <Setter Property="ItemsSource" Value="{Binding Targets}"/>
+            <Setter Property="ItemContainerStyle" Value="{StaticResource TargetMenuItemStyle}"/>
+            <Setter Property="IsCheckable" Value="False"/>
+            <Setter Property="Foreground" Value="{DynamicResource {x:Static vsui:EnvironmentColors.CommandBarTextActiveBrushKey}}"/>
+            <Setter Property="Background" Value="Transparent"/>
+            <Setter Property="AutomationProperties.Name" Value="{Binding AutomationName}"/>
+        </Style>
+
+        <!-- The key is referenced from the code behind -->
+        <!-- Style to show multiple members, when click the button it first pops a menu for all the members on this line (first level menu),
+        then each member has a second level of context menu, show its targerts -->
+        <Style x:Key="MultipleMembersContextMenuStyle" TargetType="ContextMenu">
+            <Setter Property="ItemContainerStyle" Value="{StaticResource MemberMenuItemStyle}"/>
+            <Setter Property="ItemsSource" Value="{Binding Members}"/>
+            <Setter Property="StaysOpen" Value="True"/>
+        </Style>
+
+    </Button.Resources>
+
+    <!-- Control template only shows the image -->
+    <Button.Template>
+        <ControlTemplate>
+            <imaging:CrispImage Moniker="{Binding Path=ImageMoniker}"/>
+        </ControlTemplate>
+    </Button.Template>
+
+    <Button.ContextMenu>
+        <!-- Ensure the context menu is not null -->
+        <ContextMenu/>
+    </Button.ContextMenu>
+</Button>
```

Comment:
i need a xaml person here.  i can't review this :)

if we are using xaml though, can we have classified items in the list?  that might be nice.  i'd at least like to see how that looked :)

Reply:
I have post some picture to help understanding the look :)